Sources/ApolloCodegenLib/TemplateString+AvailabilityDeprecated.swift

```diff
@@ -6,25 +6,56 @@ extension TemplateString.StringInterpolation {
   ) {
     guard
       config.options.warningsOnDeprecatedUsage == .include,
-      let deprecationReason = deprecationReason
+      let escapedDeprecationReason = deprecationReason?.escapedSwiftStringSpecialCharacters()
     else {
       removeLineIfEmpty()
       return
     }
 
-    let deprecationReasonLines = deprecationReason
-      .split(separator: "\n", omittingEmptySubsequences: false)
+    appendInterpolation("""
+      @available(*, deprecated, message: \"\(escapedDeprecationReason)\")
+      """)
+  }
 
-    if deprecationReasonLines.count > 1 {
-      appendInterpolation("""
-        @available(*, deprecated, message: \"\"\"
-          \(deprecationReasonLines.joinedAsLines(withIndent: "  "))
-          \"\"\")
-        """)
-    } else {
-      appendInterpolation("""
-        @available(*, deprecated, message: \"\(deprecationReason)\")
-        """)
+  mutating func appendInterpolation(
+    field: String,
+    argument: String,
+    warningReason: String
+  ) {
+    let escapedWarningReason = warningReason.escapedSwiftStringSpecialCharacters()
+
+    appendInterpolation("""
+      #warning("Argument '\(argument)' of field '\(field)' is deprecated. \
+      Reason: '\(escapedWarningReason)'")
+      """)
+  }
+}
+
+extension String {
+  /// Replaces specific escaped characters so they are written into the rendered deprecation
+  /// message as escaped characters to be correctly rendered to the user in an Xcode warning
+  /// (e.g., `\"` becomes `\\\"`).
+  ///
+  /// String literals can include the following special characters: `\0` (null character),
+  /// `\\` (backslash), `\t` (horizontal tab), `\n` (line feed), `\r` (carriage return),
+  /// `\"` (double quotation mark) and `\'` (single quotation mark).
+  func escapedSwiftStringSpecialCharacters() -> String {
+    var escapedString = String()
+    escapedString.reserveCapacity(self.count)
+
+    forEach { character in
+      switch (character) {
+      case "\0": escapedString.append(#"\0"#)
```

Comment:
I'm open to alternatives that would be more efficient or better performing than this; all the other ways I tried I couldn't use extended string delimiters and using regular string literals causes the escaped characters to be 'processed' instead.

Reply:
It feels like there has got to be a better way of doing this, but I'm not sure what that would be honestly, so lets go with it. This is not the most performance sensitive code to be honest.

I wonder if using a dictionary for the find and replaces would be slightly more performant, but if I remember correctly this switch statement should end up compiling into something very similar to that anyways. (I could be totally wrong about that, it's been a long time since I looked into that.)